Modules/Utils/Sources/PocketCastsUtils/General/KPMSearch.swift

```diff
@@ -2,13 +2,12 @@ import Foundation
 
 // An implementation of the Knuth-Morris-Pratt algorithm
 public class KMPSearch {
-    private var pattern: [Character]
-    private var lps: [Int]
+    private var pattern: [Character] = []
+    private var lps: [Int] = []
+    private var textArray: [String.Element]
 
-    public init(pattern: String) {
-        self.pattern = Array(pattern.lowercased())
-        self.lps = Array(repeating: 0, count: pattern.count)
-        computeLPSArray()
+    public init(text: String) {
+        textArray = Array(text.lowercased().folding(options: .diacriticInsensitive, locale: nil).replacingOccurrences(of: "ł", with: "l"))
```

Comment:
Nit:
can we wrap `.replacingOccurrences(of: "ł", with: "l")` into an Array extension?

Reply:
Done in e665abeb9174d03eb47968062d3798c549b5e70d